models/evc.py

```diff
@@ -1058,32 +1082,105 @@ def run_sdntrace(uni):
+            circuit_data[interface_z] = {
+                                            'circuit_id': circuit.id,
+                                            'trace_name': 'trace_z'
+                                        }
+        traces = EVCDeploy.run_bulk_sdntraces(uni_list)
+        del uni_list
+        circuit_by_traces = {}
+        circuits_checked = {}
+
+        for trace_switch in traces:
+            for trace in traces[trace_switch]:
+                id_trace = str(trace[0]['dpid']) + ':' + str(trace[0]['port'])
+                circuit_from_data = circuit_data.get(id_trace)
+                if circuit_from_data is None:
+                    continue
+                circuit_id = circuit_from_data['circuit_id']
+                trace_name = circuit_from_data['trace_name']
+                if circuit_id not in circuit_by_traces:
+                    circuit_by_traces[circuit_id] = {}
+                circuit_by_traces[circuit_id][trace_name] = trace
+                if 'trace_a' in circuit_by_traces[circuit_id] \
+                        and 'trace_z' in circuit_by_traces[circuit_id]:
+                    circuits_checked[circuit.id] = True
+                    trace_a = circuit_by_traces[circuit_id]['trace_a']
+                    trace_z = circuit_by_traces[circuit_id]['trace_z']
+                    if len(trace_a) != len(circuit.current_path) + 1:
+                        log.warning(f"Invalid trace from uni_a: {trace_a}")
+                        circuits_checked[circuit.id] = False
+                    if len(trace_z) != len(circuit.current_path) + 1:
+                        log.warning(f"Invalid trace from uni_z: {trace_z}")
+                        circuits_checked[circuit.id] = False
+                    for link, trace1, trace2 in zip(circuit.current_path,
+                                                    trace_a[1:],
+                                                    trace_z[:0:-1]):
+                        if compare_endpoint_trace(
+                                                    link.endpoint_a,
+                                                    glom(
+                                                            link.metadata,
+                                                            's_vlan.value'),
+                                                    trace2
+                                                ) is False:
+                            log.warning(f"Invalid trace from uni_a: {trace_a}")
+                            circuits_checked[circuit.id] = False
+                        if compare_endpoint_trace(
+                                                    link.endpoint_b,
+                                                    glom(
+                                                            link.metadata,
+                                                            's_vlan.value'),
+                                                    trace1
```

Comment:
Gretel, could you please refactor this code and maybe split it into another function (that checks the individual trace) to improve code readability?